libraries/botbuilder/src/cloudAdapter.ts

```diff
@@ -145,6 +144,35 @@ export class CloudAdapter extends CloudAdapterBase implements BotFrameworkHttpAd
         }
     }
 
+    /**
+     * Exposes the same functionality as [BotFrameworkAdapter.processActivityDirect](https://github.com/microsoft/botbuilder-js/blob/main/libraries/botbuilder/src/botFrameworkAdapter.ts).
+     *
+     * @param authorization Authorization in format: "Bearer [longString]".
+     * @param activity The activity to process.
+     * @param logic The logic function to apply.
+     * @returns a promise representing the asynchronous operation.
+     */
+    async processActivityDirect(
+        authorization: String,
+        activity: Activity,
+        logic: (context: TurnContext) => Promise<void>
+    ): Promise<void> {
+        const validatedActivity = validateAndFixActivity(ActivityT.parse(activity));
+
+        if (!validatedActivity.type) {
+            console.warn('BadRequest: Missing activity or activity type.');
+            throw new Error('Missing activity or activity type');
+        }
```

Comment:
This validation seems unnecessary. It was meant to validate the request's body but here you are receiving the activity instead.